sql/core/src/main/scala/org/apache/spark/sql/jdbc/DB2Dialect.scala

```diff
@@ -29,4 +29,8 @@ private object DB2Dialect extends JdbcDialect {
     case BooleanType => Option(JdbcType("CHAR(1)", java.sql.Types.CHAR))
     case _ => None
   }
+
+  override def maskCredentials(url: String): String = {
+    url.replaceAll("(user|password)=(.*?)(;|$)", "$1=***$3")
```

Comment:
Nit: you don't need the second capturing group, unless I'm missing something about the regex